client/src/pages/ScanStore.tsx

```diff
@@ -47,39 +91,259 @@ function ScanStore() {
     setShoppingList(sList);
   };
 
+  const setDebugItem = (barcode: string) => {
+    setCurBarcode(barcode);
+    setShoppingList(cartItems);
+  };
+
+  const addItem = async () => {
+    if (curBarcode == "") {
+      // If empty, use media API
+      const imgReq = {
+        allowedMimeTypes: ["image/jpeg"],
+        allowedSources: ["camera"], // Restrict to camera scanning only
+      };
+      const imgRes = await window.microapps
+        .requestMedia(imgReq)
+        .then((res: any) => res);
+      setUploadImg(imgRes);
+    } else {
+      let barcodes: string[] = [curBarcode];
+      fetchItem(barcodes);
+    }
+  };
+
+  const fetchItem = async (barcodes: string[]) => {
+    // Check first if we have previously retrieved this item
+    let unknown_barcodes: string[] = [];
+    let matchedItems: Item[] = [];
+    for (let i = 0; i < barcodes.length; ++i) {
+      if (idxMap[barcodes[i]] != undefined) {
+        let matchedCartItem = cartItems[idxMap[barcodes[i]]];
+        matchedItems.push(matchedCartItem.item);
+      } else {
+        unknown_barcodes.push(barcodes[i]);
+      }
+    }
+    if (matchedItems.length > 0) {
+      displayItems(matchedItems);
+    } else if (unknown_barcodes.length > 0) {
+      let data = {
+        "merchant-id": merchantID,
+        barcode: unknown_barcodes,
+      };
+      const items = await fetchJson(data, "/api/items");
+      displayItems(items);
+    }
+  };
+
+  const displayItems = (extractedItems: Item[]) => {
+    for (let i = 0; i < extractedItems.length; ++i) {
```

Comment:
Prefer for-of instead go/js-style#features-for-loops

Reply:
#30